test/integration/shoots/vpntunnel/vpntunnel.go

```diff
@@ -0,0 +1,212 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vpntunnel
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/test/framework"
+	"github.com/gardener/gardener/test/framework/resources/templates"
+
+	"github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	deploymentName = "logging-pod"
+	namespace      = "default"
+	logsCount      = 100000
+	logsDuration   = "1s"
+	loggerAppLabel = "vpnTunnelTesting"
+	testTimeout    = 5 * time.Minute
+	cleanupTimeout = 5 * time.Minute
+	kubecfgPath    = "/tmp/tunnel-test-kubecfg"
+	maxIterations  = 300
+	copyDeployment = "copy-pod"
+	copyLabel      = "vpnTunnelCopyTesting"
+)
+
+var _ = ginkgo.Describe("Shoot vpn tunnel testing", func() {
+
+	f := framework.NewShootFramework(nil)
+
+	f.Beta().CIt("should get container logs from logging-pod", func(ctx context.Context) {
+
+		ginkgo.By("Deploy the logging-pod")
+		loggerParams := map[string]interface{}{
+			"LoggerName":          deploymentName,
+			"HelmDeployNamespace": namespace,
+			"AppLabel":            loggerAppLabel,
+			"LogsCount":           logsCount,
+			"LogsDuration":        logsDuration,
+		}
+
+		err := f.RenderAndDeployTemplate(ctx, f.ShootClient, templates.VPNTunnelDeploymentName, loggerParams)
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Wait until logging-pod application is ready")
+		loggerLabels := labels.SelectorFromSet(labels.Set(map[string]string{
+			"app": loggerAppLabel,
+		}))
+
+		err = f.WaitUntilDeploymentsWithLabelsIsReady(ctx, loggerLabels, namespace, f.ShootClient)
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Get kubeconfig and extract token")
+		err = framework.DownloadKubeconfig(ctx, f.SeedClient, f.ShootSeedNamespace(), "kubecfg", kubecfgPath)
+		defer os.Remove(kubecfgPath)
+		framework.ExpectNoError(err)
+		data, err := ioutil.ReadFile(kubecfgPath)
+		framework.ExpectNoError(err)
```

Comment:
I think it's easier to not write the kubeconfig to the filesystem.
```suggestion
		data, err := GetObjectFromSecret(ctx, client, namespace, name, KubeconfigSecretKeyName)
	         if err != nil {
		  return err
	         }
```

Reply:
alternativly you can also get the token from the provided shoot client:
```
f.ShootClient.RESTConfig().BearerToken
```

Reply:
Where does the ShootClient get initialized, we have an empty restConfig here. Do we need to pass a Config to `NewShootFramework`?

Reply:
Then I think we don't gain anything.

Reply:
The shootclient is initialized [here](https://github.com/gardener/gardener/blob/master/test/framework/shootframework.go#L227) where the default `kubernetes.NewClientFromSecret` is used so the `Config should be always set`

Reply:
Kubeconfig is now stored in memory instead of downloaded.
Unfortunately the RestConfig does not contain the bearer token therefore we will leave it as it is.